src/main/java/com/google/devtools/build/lib/skylarkbuildapi/SkylarkActionFactoryApi.java

```diff
@@ -159,17 +159,70 @@
   @SkylarkCallable(
       name = "symlink",
       doc =
-          "<p><b>Experimental</b>. This parameter is experimental and may change at any "
-              + "time. Please do not depend on it. It may be enabled on an experimental basis by "
-              + "setting <code>--experimental_allow_unresolved_symlinks</code></p><p>"
-              + "Creates a symlink in the file system. If the output file is a regular file, the "
-              + "symlink must point to a file. If the output is an unresolved symlink, a dangling "
-              + "symlink is allowed.",
+          "Creates a symlink in the file system."
+              + "<p>If the output file is a regular file (i.e. created by "
+              + "<a href=\"#declare_file\"><code>declare_file()</code></a>), the symlink must "
+              + "point to a file. If the output is an unresolved symlink (i.e. created by "
+              + "<a href=\"#declare_symlink\"><code>declare_symlink()</code></a>), a dangling "
+              + "symlink is allowed.</p>",
       parameters = {
-        @Param(name = "output", type = FileApi.class, doc = "The output path.", named = true),
-        @Param(name = "target", type = String.class, doc = "The target.", named = true),
+        @Param(
+            name = "output",
+            type = FileApi.class,
+            named = true,
+            positional = false,
+            doc = "The output of this action."),
+        @Param(
+            name = "target_file",
+            type = FileApi.class,
+            named = true,
+            positional = false,
+            noneable = true,
+            defaultValue = "None",
+            doc =
+                "The target of the symlink."
+                    + "<p>If this parameter is set, <code>target_path</code> "
+                    + "must be <code>None</code>.</p>"),
+        @Param(
+            name = "target_path",
+            type = String.class,
+            named = true,
+            positional = false,
+            noneable = true,
+            defaultValue = "None",
+            doc =
+                "The target path of the symlink."
+                    + "<p>If this parameter is set, <code>target_file</code> "
+                    + "must be <code>None</code>.</p>"
+                    + "<p><b>Experimental</b> This parameter is experimental and may change at "
+                    + "any time. Please do not depend on it. It may be enabled on an experimental "
+                    + "basis by setting <code>--experimental_allow_unresolved_symlinks</code></p>"),
+        @Param(
+            name = "is_executable",
+            type = Boolean.class,
+            named = true,
+            positional = false,
+            defaultValue = "False",
+            doc =
+                "Whether the output file should be executable. "
+                    + "<p>If this is <code>True<code>, <code>target_path</code> must also be "
+                    + "executable.</p>"),
```

Comment:
`target_path` should be `target_file` here. I'd rewrite as:

> May only be used with `target_file`, not `target_path`. If true, when the action is executed, the `target_file`'s path is checked to confirm that it is executable, and an error is reported if it is not. Setting `is_executable` to `False` does not mean the target is necessarily not executable, just that no verification is done. This feature does not make sense for `target_path` because dangling symlinks might not exist at build time.

@lberki please verify what I wrote makes sense.

Reply:
Yep, exactly. Do I understand correctly that the only concern was that one instance of `target_file` was accidentally `target_path`?

Reply:
The concern was the typo and providing enough detail. I just wanted to be sure the additional detail I added wasn't blatantly wrong. :)